### DISCUSSION TITLE
5. Soruda Binary Search Kullanımı

### DISCUSSION THREAD

**Discussion by mesutckr** [gusanmaz/BMB212-Algorithms-2223#10] [Q&A] [answered]
Merhaba,
5. soruyu binary search kullanarak çözümlememizi istemişsiniz. binary search  kullanmak sıralı ve benzersiz alanların aramalarında uygun. fakat öğrenci ortalaması benzersiz bir alan değil ve aynı ortalamada belki de 5 öğrenci olacak. yazdığımız kodlarda bu durumu göz ardı edelim mi yoksa aynı ortalamada fazla kişi olabileceğini dikkate alıp kodlarımızı ona göre mi yazalım?

**Comment by gusanmaz** [comment, marked as answer]
Dedigin gibi ayni ortalamada pek cok ogrenci olabilir. Ornek data dosyalarinda da bunu gorebilirsiniz. Binary search algoritmasi illa tekrarsiz veri uzerinde calismak zorunda degil. Bu odevde klasik binary search algoritmasini size verilen soruna gore modifiye etmeniz gerekecek.
